### DISCUSSION TITLE
Are there any plans to add a Chemical Markup Language Parser?

### DISCUSSION THREAD

**Discussion by deadlyvices** [rdkit/rdkit#5780] [Ideas]
As far as I can see, RDKit supports outputting CML but not inputting it. It would be really nice for us to use RDKit on our project https://chem4word.co.uk. We make heavy use of CML throughout. 
Adding a CML parser would also help drive adoption of this comprehensive and expressive semantic format.

**Comment by greglandrum** [comment]
@deadlyvices At the moment there are no plans to support CML. It's a big format to support and doesn't enjoy particularly broad adoption, so it's never been a priority for us.

**Comment by deadlyvices** [comment]
OK, but don't you think that the reason why it hasn't had broad adoption is
because people don't support it? There are two vicious semicircles here.
Also, you don't have to support all of it. Just molecules and reactions
would be a start. The rest of it can follow as interest builds.
I'm a keen CML fan because

   1. It's free
   2. It's vendor independent and based on open standards
   3. We use it in our project (Chem4Word) - and we just support a
   rudimentary set of functions, enough to get by
   4. It can be mined by any XML capable tool, not just chemical ones
   5. It's extensible and expressive

It's gone beyond someone  taking a lead here . I think in our project we've
shown the way. I just wish others would follow.

On Tue, 22 Nov 2022 at 17:46, Greg Landrum ***@***.***> wrote:

> @deadlyvices <https://github.com/deadlyvices> At the moment there are no
> plans to support CML. It's a big format to support and doesn't enjoy
> particularly broad adoption, so it's never been a priority for us.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rdkit/rdkit/discussions/5780#discussioncomment-4208828>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACM3QMTRYSHHICSH6EAFAQ3WJUBGTANCNFSM6AAAAAASHXCRPI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Clyde
